### ISSUE TITLE
SwiftUI compatibility

### ISSUE THREAD

**Issue by afern247** [nsagora/bundle-info-versioning#5]
Does this works with SwiftUI? I added the dependency but it doesn't work, for example:
```

import BundleInfoVersioning

let bundleInfoVersioning = BundleInfoVersioning()

bundleInfoVersioning.check(forKeyPath: "CFBundleShortVersionString") { _ , newVersion in
    self.showWhatsNew(in: newVersion)
}   

```

That gives me an error on `bundleInfoVersioning.check` line saying: `Generic parameter 'T' could not be inferred`

**Comment by alexcristea** [comment]
Hi @afern247, thank you for reporting this. I will look into it to see what's happened. 

Can you help me with more details, like swift version, xcode version or maybe the signatures of the functions? For me to reproduce your setup. Thank you!

**Comment by afern247** [comment]
> Hi @afern247, thank you for reporting this. I will look into it to see what's happened.
> 
> Can you help me with more details, like swift version, xcode version or maybe the signatures of the functions? For me to reproduce your setup. Thank you!

Swift version 5.2.4
Xcode 11.5 (11E608c)
SwiftUI

The setup is very simple. A plain setup should do using the SPM

**Comment by alexcristea** [comment]
@afern247, can you look in the swift-ui example available in #6? Please note that the closure for the `check` method has two optionals:

```
bundleInfoVersioning.check(forKeyPath: "CFBundleShortVersionString") { (_ , newVersion: String?) in
    self.showWhatsNew(in: newVersion)
} 

private func showWhatsNew(in version: String?) { /* ... */}  
``` 
thus `showWhatsNew(in:)` should expect a `String?`. I will update the documentation to reflect this.

**Comment by alexcristea** [comment]
@afern247 I hope you managed to make it work. I will close the issue for the moment. We can reopen it if you still an issue.
